test/e2e/tests/confirmations/transactions/increase-token-allowance-redesign.spec.ts

```diff
@@ -0,0 +1,214 @@
+          await openDAppWithContract(driver, contractRegistry, smartContract);
+
+          await createAndAssertIncreaseAllowanceSubmission(driver, '3');
+        },
+      );
+    });
+
+    it('Sends a type 2 transaction (EIP1559) with a small spending cap', async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .withPreferencesController({
+              preferences: {
+                redesignedConfirmationsEnabled: true,
+                isRedesignedConfirmationsDeveloperEnabled: true,
+              },
+            })
+            .build(),
+          ganacheOptions: defaultGanacheOptionsForType2Transactions,
+          smartContract,
+          testSpecificMock: mocks,
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver, contractRegistry }: TestSuiteArguments) => {
+          await openDAppWithContract(driver, contractRegistry, smartContract);
+
+          await createAndAssertIncreaseAllowanceSubmission(driver, '3');
+        },
+      );
+    });
+
+    it('Sends a type 0 transaction (Legacy) with a large spending cap', async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .withPreferencesController({
+              preferences: {
+                redesignedConfirmationsEnabled: true,
+                isRedesignedConfirmationsDeveloperEnabled: true,
+              },
+            })
+            .build(),
+          ganacheOptions: defaultGanacheOptions,
+          smartContract,
+          testSpecificMock: mocks,
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver, contractRegistry }: TestSuiteArguments) => {
+          await openDAppWithContract(driver, contractRegistry, smartContract);
+
+          await createAndAssertIncreaseAllowanceSubmission(driver, '3000');
+        },
+      );
+    });
+
+    it('Sends a type 2 transaction (EIP1559) with a large spending cap', async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .withPreferencesController({
+              preferences: {
+                redesignedConfirmationsEnabled: true,
+                isRedesignedConfirmationsDeveloperEnabled: true,
+              },
+            })
+            .build(),
+          ganacheOptions: defaultGanacheOptionsForType2Transactions,
+          smartContract,
+          testSpecificMock: mocks,
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver, contractRegistry }: TestSuiteArguments) => {
+          await openDAppWithContract(driver, contractRegistry, smartContract);
+
+          await createAndAssertIncreaseAllowanceSubmission(driver, '3000');
```

Comment:
[suggestion] we could combine openDAppWithContract into createAndAssertIncreaseAllowanceSubmission where
createAndAssertIncreaseAllowanceSubmission would accept 3 params: driver, contractRegistry, newSpendingCap